File: Abstract Data Types/WeakMap.py
import math
import weakref

_data = weakref.WeakKeyDictionary()


class Circle:
    def __init__(self, radius):
        _data[self] = {"radius": radius}

    def area(self):
        radius = _data[self]["radius"]
        return math.pi * radius ** 2

    @property
    def radius(self):
        return _data[self]["radius"]


c = Circle(7)
print(c.area())
print(c.radius)

File: Abstract Data Types/WeakMap.js
const _data = new WeakMap();

class Circle {
  constructor(radius) {
    _data.set(this, { radius });    // private storage
  }

  area() {
    const { radius } = _data.get(this);
    return Math.PI * radius ** 2;
  }

  get radius() { return _data.get(this).radius; }
}

const c = new Circle(7);
console.log(c.area());   // 153.94...
console.log(c.radius);   // 7